### PR TITLE
feat(python): use minimum version for pip packages

### LINKED ISSUE #7300: feat(pip): use minimum version in range
## Description
If it is difficult to identify the version in `requirements.txt`, the minimum version should be used with [--detection-priority comprehensive](https://github.com/aquasecurity/trivy/issues/7270)

```
keyring >= 4.1.1            # Minimum version 4.1.1
Mopidy-Dirble ~= 1.1        # Compatible release. Same as >= 1.1, == 1.*
```

In the above example, the versions should be `4.1.1` and `1.1`.

### PR DESCRIPTION
**Pull request by afdesk**
## Description

This PR adds a minimum version detection for [Compatible release](https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release) and [Inclusive ordered comparison](https://packaging.python.org/en/latest/specifications/version-specifiers/#inclusive-ordered-comparison) clauses inside pip packages.

It works only with [--detection-priority comprehensive](https://github.com/aquasecurity/trivy/issues/7270).

## Refs
* Python version specifiers: https://packaging.python.org/en/latest/specifications/version-specifiers/#id5

## Related issues
- Close #7300 

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/dependency/parser/python/pip/parse.go [hunk 1]
```diff
@@ -25,15 +25,30 @@ const (
...
-func NewParser() *Parser {
+func NewParser(useMinVersion bool) *Parser {
 	return &Parser{
-		logger: log.WithPrefix("pip"),
+		logger:        log.WithPrefix("pip"),
+		useMinVersion: useMinVersion,
 	}
 }
 
+func splitLine(line string) []string {
```
Comment (DmitriyLewen) [inline pkg/dependency/parser/python/pip/parse.go:39]: I thought about `*`.
```
== 3.1.*: any version that starts with 3.1. Equivalent to the ~=3.1.0 compatible release clause.
```

We show minimal version, so it looks like we can replace `*` to `0`.
wdyt?

File: pkg/dependency/parser/python/pip/parse.go [hunk 2]
```diff
@@ -25,15 +25,30 @@ const (
...
+	if len(result) == 2 {
+		return result
+	}
+	result = strings.Split(line, ">=")
+	if len(result) == 2 {
+		return result
+	}
+
+	return strings.Split(line, "==")
+}
```
Comment (DmitriyLewen) [inline pkg/dependency/parser/python/pip/parse.go:50]: wdyt?
```suggestion
func (p *Parser) splitLine(line string) []string {
	separators := []string{"~=", ">=", "=="}
	// Without useMinVersion check only `==`
	if !p.useMinVersion {
		separators = []string{"=="}
	}
	for _, sep := range separators {
		if result := strings.Split(line, sep); len(result) == 2 {
			return result
		}
	}
	return nil
}
```

**Review by DmitriyLewen — COMMENTED** [review]
@afdesk I left couple comments.

Can you also update [python page](https://github.com/aquasecurity/trivy/blob/main/docs/docs/coverage/language/python.md) in docs?

File: pkg/dependency/parser/python/pip/parse.go (hunk 1 above)
Comment (knqyf263) [reply pkg/dependency/parser/python/pip/parse.go:39]: Agree!

File: pkg/dependency/parser/python/pip/parse.go (hunk 2 above)
Comment (afdesk) [reply pkg/dependency/parser/python/pip/parse.go]: done

File: pkg/dependency/parser/python/pip/parse.go (hunk 1 above)
Comment (afdesk) [reply pkg/dependency/parser/python/pip/parse.go]: replacing a trailing `*` is added.

File: pkg/dependency/parser/python/pip/parse.go [hunk 3]
```diff
@@ -53,10 +68,14 @@ func (p *Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependenc
 		line = rStripByKey(line, commentMarker)
 		line = rStripByKey(line, endColon)
 		line = rStripByKey(line, hashMarker)
-		s := strings.Split(line, "==")
+
+		s := p.splitLine(line)
 		if len(s) != 2 {
 			continue
 		}
+		if p.useMinVersion && strings.Index(s[1], ".*") > -1 {
```
Comment (DmitriyLewen) [inline pkg/dependency/parser/python/pip/parse.go:76]: IIUC we can check `.*` as suffix:
```bash
root@5c86285e4f39:/app# pip install django==5.*.*
ERROR: Invalid requirement: 'django==5.*.*': .* suffix can only be used with `==` or `!=` operators
    django==5.*.*
          ~~~~~~^
root@5c86285e4f39:/app# pip install django==4.*.1
ERROR: Invalid requirement: 'django==4.*.1': Expected end or semicolon (after version specifier)
    django==4.*.1
          ~~~~~^
root@5c86285e4f39:/app# pip install django==5.*
Collecting django==5.*
  Downloading Django-5.1-py3-none-any.whl.metadata (4.2 kB)
```

Comment (DmitriyLewen) [reply pkg/dependency/parser/python/pip/parse.go:76]: Do we need this check (I mean `strings.Index(s[1], ".*") > -1`)?
It seems like we can always replace the `.*` suffix if `p.useMinVersion==true`.
Or am I missing something?

Comment (afdesk) [reply pkg/dependency/parser/python/pip/parse.go]: @DmitriyLewen ok, removed this check

Comment (afdesk) [reply pkg/dependency/parser/python/pip/parse.go]: no, it doesn't work correctly. updating.

**Review by DmitriyLewen — APPROVED** [review]
LGTM

**Pull request merged**